### ISSUE TITLE
Prefer zlib over zopfli for gzip?

### ISSUE THREAD

**Issue by keeganstreet** [Alorel/shrink-ray#39]
There was some nice work done in #36 to make `iltorb` and `node-zopfli-es` optional.

But [zopfli-compat.js](https://github.com/Alorel/shrink-ray/blob/master/zopfli-compat.js) is preferring `zopfli` over `zlib`, and printing a warning if `node-zopfli-es` is not installed. `zopfli` may create gzip assets 5% smaller than `zlib`, but it is [much slower](https://www.npmjs.com/package/node-zopfli-es) (~100x).

Since `shrink-ray-current` is for use at runtime, speed is very important and I would prefer to use `zlib` not `zopfli`. This would be consistent with [brotli-compat.js](https://github.com/Alorel/shrink-ray/blob/master/brotli-compat.js) which is preferring `zlib` over `iltorb`. Would you be open to this change, or alternatively silencing the warning `Module "node-zopfli-es" was unavailable`?

**Comment by Alorel** [comment]
Hi and sorry for the late reply.

I would be more keen on silencing the error or adding a config option for it. node-zopfli does tend to offer better compression than zlib, even if slightly, and I believe it should be favoured in most scenarios where you're not trying to squeeze out the most CPU savings as possible.

**Comment by keeganstreet** [comment]
Hi Alorel, that's OK. If we add an option the change might look something like this:

- Add a boolean option called `useZopfliForGzip` which defaults to `true`
- In index.js, move `const zopfli = zopfliCompat();` inside the [`compression` function](https://github.com/Alorel/shrink-ray/blob/master/index.js#L75) so we can pass `opts.useZopfliForGzip` to `zopfliCompat`.
- In `zopfli-comat.js`, `return require('zlib')` if `useZopfliForGzip` is false.

Would you like a PR for this?

**Comment by Alorel** [comment]
Yep, that looks good. And yes, if you'd be willing to make a PR that'd be amazing, else I could get round to it over the weekend.

**Comment by keeganstreet** [comment]
I won't have time this week, but this is one of my next cards to pick up next week at work. No need to work on the weekend 😄

**Comment by CodeIter** [comment]
ok travis-ci test **passed** but **COVERAGE** _DECREASED_ (**-0.5%**) TO **98.626%** in file `zopfli-compat.js`

**Comment by CodeIter** [comment]
**COVERAGE** _DECREASED_ (**-0.3%**) TO **98.898%** in file `zopfli-compat.js`
```js
//...
function getZopfliModule(useZopfliForGzip) {
    try {
      if(useZopfliForGzip){
        return require('node-zopfli-es');
      }
//...
```

**Comment by CodeIter** [comment]
i can't get 100% COVERAGE . it 's up to you 😄

**Comment by Alorel** [comment]
Thanks for the contribution, published as 4.1.0
